app/hotels/src/filter/price/PricePopup.js

```diff
@@ -14,105 +14,109 @@ import {
 } from '@kiwicom/mobile-shared';
 import { defaultTokens } from '@kiwicom/mobile-orbit';
 import { SafeAreaView } from 'react-navigation';
+import { Decimal } from 'decimal.js-light';
 
 type Props = {|
   onClose: () => void,
   onSave: ({ minPrice: number, maxPrice: number }) => void,
   isVisible: boolean,
-  min: number,
-  max: number,
-  start: number,
-  end: number,
+  min: Decimal,
+  max: Decimal,
+  start: Decimal,
+  end: Decimal,
   currency: string,
   daysOfStay: number,
 |};
 
-type State = {|
-  price: {
-    start: number,
-    end: number,
-  },
-|};
-
-export default class PricePopup extends React.Component<Props, State> {
-  state = {
-    price: {
-      start: 0,
-      end: 0,
-    },
-  };
+export default function PricePopup({
+  min,
+  max,
+  currency,
+  daysOfStay,
+  onSave,
+  end,
+  start,
+  onClose,
+  isVisible,
+}: Props) {
+  const [price, setPrice] = React.useState({
+    start: new Decimal(0),
+    end: new Decimal(0),
+  });
 
-  static getDerivedStateFromProps = ({ start, end, isVisible }: Props) => {
-    return isVisible
-      ? null
-      : {
-          price: {
-            start,
-            end,
-          },
-        };
-  };
-
-  handlePriceChanged = ([start, end]: number[]) =>
-    this.setState({
-      price: { start, end },
+  React.useEffect(() => {
+    setPrice({
+      start,
+      end,
     });
+  }, [end, start]);
 
-  onSave = () =>
-    this.props.onSave({
-      minPrice: this.state.price.start,
-      maxPrice: this.state.price.end,
-    });
+  const handlePriceChanged = React.useCallback(
+    ([start, end]: number[]) =>
+      setPrice({
+        start: new Decimal(start),
+        end: new Decimal(end),
+      }),
+    [],
+  );
 
-  render() {
-    const { start, end } = this.state.price;
-    const { min, max, currency, daysOfStay } = this.props;
+  const savePrice = React.useCallback(() => {
```

Comment:
Can you explain why `useCallback` is used here? And why not use `useMemo` or nothing?

Reply:
> The useCallback Hook lets you keep the same callback reference between re-renders so that shouldComponentUpdate continues to work

> The useMemo Hook makes it easier to control when individual children update, reducing the need for pure components.

>  useCallback(fn, deps) is equivalent to useMemo(() => fn, deps)

I guess both useCallback and useMemo could be used here.